tests/ui/consts/const-eval/ub-enum.32bit.stderr

```diff
@@ -108,13 +108,13 @@ error[E0080]: evaluation of constant value failed
   --> $DIR/ub-enum.rs:96:77
    |
 LL | const BAD_UNINHABITED_WITH_DATA1: Result<(i32, Never), (i32, !)> = unsafe { mem::transmute(0u64) };
-   |                                                                             ^^^^^^^^^^^^^^^^^^^^ transmuting to uninhabited type
+   |                                                                             ^^^^^^^^^^^^^^^^^^^^ entering unreachable code
 
 error[E0080]: evaluation of constant value failed
   --> $DIR/ub-enum.rs:98:77
    |
 LL | const BAD_UNINHABITED_WITH_DATA2: Result<(i32, !), (i32, Never)> = unsafe { mem::transmute(0u64) };
-   |                                                                             ^^^^^^^^^^^^^^^^^^^^ transmuting to uninhabited type
+   |                                                                             ^^^^^^^^^^^^^^^^^^^^ entering unreachable code
```

Comment:
Ok, these are back to being "unreachable code" now, not mentioning `transmute` at all.

Reply:
That makes me wonder what the underlying MIR looks like -- ideally we'd evaluate to the end and then the usual validation path would give us a nice message; seems like somewhere we are producing an `Unreachable` terminator instead.

Reply:
Before lowering:
```rust
// MIR for `BAD_UNINHABITED_WITH_DATA2` before LowerIntrinsics

const BAD_UNINHABITED_WITH_DATA2: Result<(i32, !), (i32, Never)> = {
    let mut _0: std::result::Result<(i32, !), (i32, Never)>; // return place in scope 0 at C:\src\rust\tests\codegen\intrinsics\transmute.rs:12:39: 12:69
    scope 1 {
    }

    bb0: {
        _0 = transmute::<u64, Result<(i32, !), (i32, Never)>>(const 0_u64); // scope 1 at C:\src\rust\tests\codegen\intrinsics\transmute.rs:12:81: 12:106
                                         // mir::Constant
                                         // + span: C:\src\rust\tests\codegen\intrinsics\transmute.rs:12:81: 12:100
                                         // + literal: Const { ty: unsafe extern "rust-intrinsic" fn(u64) -> Result<(i32, !), (i32, Never)> {transmute::<u64, Result<(i32, !), (i32, Never)>>}, val: Value(<ZST>) }
    }

    bb1 (cleanup): {
        resume;                          // scope 0 at C:\src\rust\tests\codegen\intrinsics\transmute.rs:12:1: 12:109
    }
}
```

Later:
```rust
// MIR for `BAD_UNINHABITED_WITH_DATA2` after runtime-post-cleanup

const BAD_UNINHABITED_WITH_DATA2: Result<(i32, !), (i32, Never)> = {
    let mut _0: std::result::Result<(i32, !), (i32, Never)>; // return place in scope 0 at C:\src\rust\tests\codegen\intrinsics\transmute.rs:12:39: 12:69
    scope 1 {
    }

    bb0: {
        _0 = const 0_u64 as std::result::Result<(i32, !), (i32, Never)> (Transmute); // scope 1 at C:\src\rust\tests\codegen\intrinsics\transmute.rs:12:81: 12:106
        unreachable;                     // scope 1 at C:\src\rust\tests\codegen\intrinsics\transmute.rs:12:81: 12:106
    }
}
```

An earlier version of this PR took out the Cast if there was no `target`, but it's now kept as a statement even if it's unreachable.

I guess `copy_op` isn't checking validity?

https://github.com/rust-lang/rust/pull/108442/files#diff-ca7143bbe4e8d13d348b1dff24a00da8d0ec776462667f3fa22d63fa72f72774L415

Reply:
Marking this thread resolved now that #109435 is in the bors queue.